Crash on terraform validate warning

### Describe the bug
CTS crashes when the `terraform validate` it runs results in a warning:

```
consul-terraform-sync -config-file consul-terraform-sync-acme.hcl

2021-10-31T16:42:42.907+0100 [INFO]  cli: 0.4.0
2021-10-31T16:42:42.908+0100 [INFO]  cli: running controller in daemon mode
2021-10-31T16:42:42.908+0100 [INFO]  cli: setting up controller: type=readwrite
2021-10-31T16:42:42.908+0100 [INFO]  ctrl: initializing Consul client and testing connection
2021-10-31T16:42:42.986+0100 [INFO]  driver.terraform: skipping install, terraform already exists: tf_version=1.0.6 install_path=/home/consul-terraform-sync-le
2021-10-31T16:42:42.986+0100 [INFO]  cli: initializing controller
2021-10-31T16:42:42.986+0100 [INFO]  ctrl: initializing driver
2021-10-31T16:42:42.986+0100 [INFO]  ctrl: initializing all tasks
2021-10-31T16:42:42.987+0100 [INFO]  ctrl: initializing task: task=test
2021-10-31T16:42:42.988+0100 [INFO]  client.terraformcli: Terraform output is muted
2021-10-31T16:42:42.989+0100 [INFO]  driver.terraform: retrieved 0 Terraform handlers for task: task_name=test
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xaaf4f7]

goroutine 50 [running]:
github.com/hashicorp/consul-terraform-sync/client.(*TerraformCLI).Validate(0xc00001f4a0, 0x1650808, 0xc0004b12c0, 0x0, 0x0)
        /home/runner/work/consul-terraform-sync/consul-terraform-sync/client/terraform_cli.go:227 +0x6b7
github.com/hashicorp/consul-terraform-sync/driver.(*Terraform).validateTask(...)
        /home/runner/work/consul-terraform-sync/consul-terraform-sync/driver/terraform.go:565
github.com/hashicorp/consul-terraform-sync/driver.(*Terraform).initTask(0xc0001a10e0, 0x1650808, 0xc0004b12c0, 0x0, 0xc000036184)
        /home/runner/work/consul-terraform-sync/consul-terraform-sync/driver/terraform.go:392 +0x339
github.com/hashicorp/consul-terraform-sync/driver.(*Terraform).InitTask(0xc0001a10e0, 0x1650808, 0xc0004b12c0, 0x0, 0x0)
        /home/runner/work/consul-terraform-sync/consul-terraform-sync/driver/terraform.go:172 +0x1c5
github.com/hashicorp/consul-terraform-sync/controller.(*baseController).init(0xc000258000, 0x1650808, 0xc0004b12c0, 0x1, 0xc000072000)
        /home/runner/work/consul-terraform-sync/consul-terraform-sync/controller/controller.go:114 +0x35d
github.com/hashicorp/consul-terraform-sync/controller.(*ReadWrite).Init(0xc0004b0040, 0x1650808, 0xc0004b12c0, 0x0, 0x0)
        /home/runner/work/consul-terraform-sync/consul-terraform-sync/controller/readwrite.go:52 +0x45
github.com/hashicorp/consul-terraform-sync/command.(*CLI).runBinary.func1(0x1660668, 0xc00019b980, 0x1650aa8, 0xc0004b0040, 0x1650808, 0xc0004b12c0, 0xc0003e8060, 0xc0003e6120, 0x200000)
        /home/runner/work/consul-terraform-sync/consul-terraform-sync/command/cli.go:233 +0xbd
created by github.com/hashicorp/consul-terraform-sync/command.(*CLI).runBinary
        /home/runner/work/consul-terraform-sync/consul-terraform-sync/command/cli.go:231 +0x9d7

```

Running `terraform validate` manually on the generated configuration outputs the following:

```
terraform validate
╷
│ Warning: Empty provider configuration blocks are not required
│
│   on .terraform/modules/test/main.tf line 21:
│   21: provider "acme" {
│
│ Remove the acme provider block from module.test.
│
│ (and one more similar warning elsewhere)
╵
Success! The configuration is valid, but there were some validation warnings as shown above.
```

### Versions
**Consul Terraform Sync**
```
consul-terraform-sync 0.4.0
Compatible with Terraform >= 0.13.0, < 1.1.0
```

**Consul Version**
```
Consul v1.7.3
```

**Terraform Version**
```
Terraform v1.0.6
```

### Configuration File(s)

I think the configuration is irrelevant, but if it can help i can share it.

### Steps to Reproduce
1. Have a Terraform module which outputs a warning on `terraform validate` ( e.g. empty provider block in main.tf)
2. Run CTS 
3. Crash

--- a/client/terraform_cli.go
+++ b/client/terraform_cli.go
@@ -224,7 +224,12 @@ func (t *TerraformCLI) Validate(ctx context.Context) error {
 		default:
 			fmt.Fprintf(&sb, "%s: %s\n", d.Severity, d.Summary)
 			if d.Range != nil && d.Snippet != nil {
-				fmt.Fprintf(&sb, "\non %s line %d, in %s\n", d.Range.Filename, d.Range.Start.Line, *d.Snippet.Context)
+				if d.Snippet.Context != nil {
+					fmt.Fprintf(&sb, "\non %s line %d, in %s\n",
+						d.Range.Filename, d.Range.Start.Line, *d.Snippet.Context)
+				} else {
+					fmt.Fprintf(&sb, "\non %s line %d\n", d.Range.Filename, d.Range.Start.Line)
+				}
 				fmt.Fprintf(&sb, "%d:%s\n\n", d.Snippet.StartLine, d.Snippet.Code)
 			}
 			sb.WriteString(d.Detail)

